### PR TITLE
UNWIND an expression that is not a list

### PR DESCRIPTION
**Pull request by nafraf**
Hi, this PR is to solve the bug #2635 

The function _initList() was modified to create a list with the input value when the input type is not a list.
A new test file was created to test unwind clause.

### CODE REVIEW

File: tests/flow/test_unwind_clause.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
+from common import *
+import re
+
+redis_graph = None
+
+class testUnwindClause(FlowTestsBase):
```
Comment (swilly22) [inline tests/flow/test_unwind_clause.py:6]: I think we can drop `FlowTestsBase`
```python
class testUnwindClause():
```

File: tests/flow/test_unwind_clause.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    
+    def __init__(self):
+        self.env = Env(decodeResponses=True)
+        global redis_graph
+        redis_con = self.env.getConnection()
+        redis_graph = Graph(redis_con, "G")
```
Comment (swilly22) [inline tests/flow/test_unwind_clause.py:12]:
```python
GRAPH_ID = "unwind"
...
 redis_graph = Graph(redis_con, GRAPH_ID)
```

File: tests/flow/test_unwind_clause.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+ 
+    def test01_unwind_null_output(self):
+        query = """UNWIND ({x:null}) AS q RETURN q.x"""
```
Comment (swilly22) [inline tests/flow/test_unwind_clause.py:15]: `q` isn't NULL, it is a MAP object containing a key `x` with the value `NULL`
`UNWIND ({x:null}) AS q RETURN q` would have returned a valid map object.

File: tests/flow/test_unwind_clause.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        query = """UNWIND [] AS x RETURN x"""
+        actual_result = redis_graph.query(query)
+        expected = []
+        self.env.assertEqual(actual_result.result_set, expected)
+
+        # list including null
+        query = """UNWIND [1, 2, null] AS x RETURN x"""
+        actual_result = redis_graph.query(query)
+        expected = [[1], [2], [None]]
+        self.env.assertEqual(actual_result.result_set, expected)
```
Comment (swilly22) [inline tests/flow/test_unwind_clause.py:78]: Please add `UNWIND [1, null, 2] AS x RETURN x`
I think the iteration should halt after `NULL` and `2` won't be omitted? please double check this against Neo4J

We might want to make sure this behavior is covered by TCK

File: tests/flow/test_unwind_clause.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    def test02_unwind_set(self):
+        # delete property
+        query = """CREATE (n:N {x:3})"""
+        actual_result = redis_graph.query(query)
+        query = """UNWIND ({x:null}) AS q MATCH (n:N) SET n.x= q.x RETURN n"""
+        actual_result = redis_graph.query(query)
+        self.env.assertEqual(actual_result.properties_removed, 1)
+        query = """MATCH (n:N) DELETE n"""
+        redis_graph.query(query)
```
Comment (swilly22) [inline tests/flow/test_unwind_clause.py:88]: there's no need for this last DELETE query.

File: src/execution_plan/ops/op_unwind.c [hunk 6]
```diff
@@ -36,18 +36,20 @@ OpBase *NewUnwindOp(const ExecutionPlan *plan, AR_ExpNode *exp) {
...
 static void _initList(OpUnwind *op) {
 	op->list = SI_NullVal(); // Null-set the list value to avoid memory errors if evaluation fails.
 	SIValue new_list = AR_EXP_Evaluate(op->exp, op->currentRecord);
-	if(SI_TYPE(new_list) != T_ARRAY) {
-		Error_SITypeMismatch(new_list, T_ARRAY);
+	if(SI_TYPE(new_list) == T_ARRAY) {
+		// Update the list value.
+		op->list = new_list;
+	} else {
+		SIValue array = SI_Array(1);
```
Comment (swilly22) [inline src/execution_plan/ops/op_unwind.c:48]: Please add a comment explaining what's going on within the `else` block

File: src/execution_plan/ops/op_unwind.c [hunk 7]
```diff
@@ -36,18 +36,20 @@ OpBase *NewUnwindOp(const ExecutionPlan *plan, AR_ExpNode *exp) {
...
+		SIArray_Append(&array, new_list);
 		SIValue_Free(new_list);
-		ErrorCtx_RaiseRuntimeException(NULL);
+		op->list = array;
```
Comment (swilly22) [inline src/execution_plan/ops/op_unwind.c:51]: Maybe:
```c
op->list = SI_Array(1);
SIArray_Append(&op->list, new_list);
SIValue_Free(new_list);
```

**Review by swilly22 — CHANGES_REQUESTED** [review]
Nice work, please see comments.

File: src/execution_plan/ops/op_unwind.c (hunk 7 above)
Comment (nafraf) [reply src/execution_plan/ops/op_unwind.c]: Thanks, change done in [fb9dd7f](https://github.com/RedisGraph/RedisGraph/pull/2650/commits/fb9dd7fa3981c21067e21660caafd9ee4c04cfc6)

File: src/execution_plan/ops/op_unwind.c (hunk 6 above)
Comment (nafraf) [reply src/execution_plan/ops/op_unwind.c]: Ok. Done in [fb9dd7f](https://github.com/RedisGraph/RedisGraph/pull/2650/commits/fb9dd7fa3981c21067e21660caafd9ee4c04cfc6)

File: tests/flow/test_unwind_clause.py (hunk 5 above)
Comment (nafraf) [reply tests/flow/test_unwind_clause.py]: Fixed in [d68e6b1](https://github.com/RedisGraph/RedisGraph/pull/2650/commits/d68e6b176ba7b549700390b4ad0a81d2ad9fef2d)

File: tests/flow/test_unwind_clause.py (hunk 3 above)
Comment (nafraf) [reply tests/flow/test_unwind_clause.py]: Ok, the test was fixed.
The unwinding null test was enabled in TCK tests.

File: tests/flow/test_unwind_clause.py (hunk 4 above)
Comment (nafraf) [reply tests/flow/test_unwind_clause.py:80]: The iteration does not halt after ```NULL```, the tests were done in Neo4J.

I think that this behavior is covered by TCK test like the test in TypeConversion1 [Scenario: [4] `toBoolean()` on invalid strings](https://github.com/RedisGraph/RedisGraph/blob/ad07715bfb8c27f604bc6b2a1fb4a4bc55ac8238/tests/tck/features/expressions/typeConversion/TypeConversion1.feature#L70)
where the input list has a null at first position.

File: tests/flow/test_unwind_clause.py (hunk 2 above)
Comment (nafraf) [reply tests/flow/test_unwind_clause.py]: Ok, change done in [d68e6b1](https://github.com/RedisGraph/RedisGraph/pull/2650/commits/d68e6b176ba7b549700390b4ad0a81d2ad9fef2d). Thanks

File: tests/flow/test_unwind_clause.py (hunk 1 above)
Comment (nafraf) [reply tests/flow/test_unwind_clause.py]: Ok. Done.

**Pull request merged**